lib/ruboty/adapters/slack_socket_mode.rb

```diff
@@ -56,12 +44,31 @@ def say(message)
             initial_comment: message[:body] || ''
           )
           client.files_upload(args)
+          return
+        end
+
+        if message[:attachments] && !message[:attachments].empty?
```

Comment:
[Q] `message[:attachments]` は空文字になることがあるのでしょうか？

空文字になることがないのであれば `#empty?` はいらないと思います。
空文字になることがあるのであれば `nil` は空文字として扱うと少しシンプルになりそうです。

```suggestion
        if !message.fetch(:attachments, "").empty?
```

Reply:
このメソッドは Ruboty の Handler, Action で `message.reply` した時の引数をほぼそのまま受け継いで動きますので、
`message.reply('', attachments: '')` といった実装があれば空文字が入ることはありますね・・

実際には attachments は JSONオブジェクトの配列が想定されていると思っており、空の配列を弾きたいのかと思っています。（ここはfork元からある実装で完全な把握ができていないというのがあり、申し訳ありません。）
https://api.slack.com/reference/messaging/attachments

いずれにしても、空が来ないという保証は出来ないところなので、実装をお借りして
`if !message.fetch(:attachments, []).empty?` といった形にしてみようと思います！